tests/test_unit/test_utils.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import tempfile
+
+import pytest
+import yaml
+
+from wazp.utils import df_from_metadata_yaml_files
+
+
+def get_sample_project_metadata_fields() -> dict:
+    """Get the metadata dictionary from the sample project for testing."""
+    with open("sample_project/metadata_fields.yaml") as fi:
+        metadata_fields = yaml.safe_load(fi)
+    return metadata_fields
+
+
+def test_df_from_metadata_yaml_sample_project_metadata() -> None:
+    """Normal operation: test we can read the sample project metadata."""
+    fields = get_sample_project_metadata_fields()
+    testme = df_from_metadata_yaml_files("sample_project/videos", fields)
+
+    expected = set(fields)
+    actual = set(testme.columns)
+    diff = expected.symmetric_difference(actual)
+    assert expected == actual, f"Metadata fields -> df problem with: {diff}"
+
+    nfiles = len(glob.glob("sample_project/videos/*.yaml"))
+    nrows, _ = testme.shape
+    assert nrows == nfiles, "Number of rows in df != number of yaml files."
+
+
+def test_df_from_metadata_yaml_no_metadata() -> None:
+    """
+    Test with no metadata files (expect just to create an empty dataframe with
+    metadata_fields column headers).
+    """
+    fields = get_sample_project_metadata_fields()
+    with tempfile.TemporaryDirectory() as empty_existing_directory:
```

Comment:
nice! ✨ 
I wasn't aware of this way of making temp dirs

Reply:
I independently stumbled upon this yesterday, very neat!

Reply:
😎